index.js

```diff
@@ -1,5 +1,78 @@
-console.log("hola");
 
+"use strict";
 const mainContainer = document.querySelector("#app");
+const listOfCards = document.querySelector("#lista");
+const shuffle = document.querySelector(".shuffle");
+const topCard = document.querySelector("#top-card");
+
+const cardsArr = [];
+const clubs = ["clubs", "diamonds", "hearts", "spades"];
+const cards = ["As", 2, 3, 4, 5, 6, 7, 8, 9, 10, "J", "Q", "K"];
+
+//GENERANDO ARRAY CON CARTAS
+for (let i = 0; i < 4; i++) {
+  for (let j = 0; j < 13; j++) {
+    let cardObj = {
+      suit: clubs[i],
+      number: cards[j],
+      color: clubs[i] === "hearts" || clubs[i] === "spades" ? "red" : "black",
+    };
+
+    cardsArr.push(cardObj);
+  }
+}
+
+//FUNCION PARA REVOLVER CARTAS
+
+const revolver = function (arr) {
+  let largo = arr.length;
+  let indexAleatorio;
+
+  while (largo !== 0) {
+    //generando un numero aleatorio entre 0 y el ultimo indice
+    indexAleatorio = Math.round(Math.random() * (largo - 1));
+    largo--;
+
+    //cambiando ultimo numero con indice aleatorio
+
+    [arr[largo], arr[indexAleatorio]] = [arr[indexAleatorio], arr[largo]];
+  }
+  return arr;
+};
+
+const barajear = function () {
+  const cardsArrRevuelto = revolver(cardsArr);
+
+  let cardsListHtmlString = "";
+  let icon;
+
+  cardsArrRevuelto.forEach((card) => {
```

Comment:
We can refactor this into the card generation logic, so that the icon is assigned at the moment the cards are being generated